src/Sylius/Behat/Context/Api/Admin/ManagingExchangeRatesContext.php

```diff
@@ -0,0 +1,344 @@
+        float $ratio,
+        CurrencyInterface $sourceCurrency,
+        CurrencyInterface $targetCurrency
+    ): void {
+        Assert::true(
+            $this->hasExchangeRateFromResponse($ratio, $sourceCurrency, $targetCurrency),
+            sprintf(
+                'Exchange rate with ratio %s between %s and %s does not exist',
+                $ratio,
+                $sourceCurrency->getName(),
+                $targetCurrency->getName()
+            )
+        );
+    }
+
+    /**
+     * @Then I should see the exchange rate between :sourceCurrency and :targetCurrency in the list
+     * @Then I should (also) see an exchange rate between :sourceCurrency and :targetCurrency on the list
+     */
+    public function iShouldSeeTheExchangeRateBetweenAndInTheList(
+        CurrencyInterface $sourceCurrency,
+        CurrencyInterface $targetCurrency
+    ): void {
+        Assert::notNull($this->getExchangeRateFromResponse($sourceCurrency, $targetCurrency));
+    }
+
+    /**
+     * @Then it should have a ratio of :ratio
+     */
+    public function itShouldHaveARatioOf(float $ratio): void
+    {
+        $this->client->index('exchange_rates');
+
+        Assert::true(
+            $this->client->hasItemWithValue('ratio', $ratio),
+            sprintf('ExchangeRate with ratio %s does not exist', $ratio)
+        );
+    }
+
+    /**
+     * @Then /^(this exchange rate) should no longer be on the list$/
+     */
+    public function thisExchangeRateShouldNoLongerBeOnTheList(ExchangeRateInterface $exchangeRate): void
+    {
+        Assert::false(
+            $this->hasExchangeRateFromResponse(
+                $exchangeRate->getRatio(),
+                $exchangeRate->getSourceCurrency(),
+                $exchangeRate->getTargetCurrency()
+            ),
+            sprintf(
+                'Exchange rate with ratio %s between %s and %s still exists',
+                $exchangeRate->getRatio(),
+                $exchangeRate->getSourceCurrency()->getName(),
+                $exchangeRate->getTargetCurrency()->getName()
+            )
+        );
+    }
+
+    /**
+     * @Then the exchange rate between :sourceCurrency and :targetCurrency should not be added
+     */
+    public function theExchangeRateBetweenAndShouldNotBeAdded(
+        CurrencyInterface $sourceCurrency,
+        CurrencyInterface $targetCurrency
+    ): void {
+        $this->client->index('exchange_rates');
+
+        Assert::null($this->getExchangeRateFromResponse($sourceCurrency, $targetCurrency));
+    }
+
+    /**
+     * @Then /^(this exchange rate) should have a ratio of ([0-9\.]+)$/
+     */
+    public function thisExchangeRateShouldHaveARatioOf(ExchangeRateInterface $exchangeRate, string $ratio): void
+    {
+        $exchangeRate = $this->getExchangeRateFromResponse(
+            $exchangeRate->getSourceCurrency(),
+            $exchangeRate->getTargetCurrency()
+        );
+
+        $exchangeRate['ratio'] = $ratio;
```

Comment:
It seems that some logic is missing here. Are you sure, you wanted to assign value to the `ratio` key in array?